### PR TITLE
Linbee 6728 add option to run GitHub action with cache

### PR DESCRIPTION
**Pull request by MishaKav**
[workerB](https://workerb.linearb.io/v2/badge/collaboration-page?magicLinkId=Vol20V8)

- add ability  to pass `pull_docker_from_cache` in order to store docker in github cache

### CODE REVIEW

**Review by yeelali14 — APPROVED** [review]
lgtm

File: action.yml [hunk 1]
```diff
@@ -100,6 +104,17 @@ runs:
...
         path: "gitstream/cm/"
 
+    - name: Docker Layer Caching
+      if: ${{ inputs.pull_docker_from_cache == 'true' }}
+      uses: satackey/action-docker-layer-caching@v0.0.11
+      continue-on-error: true
+      with:
+        concurrency: 12
+
+    - name: Pull gitstream Docker
```
Comment (ShakedZrihen) [inline action.yml:114]: It will cache all dockers? How does it knows which docker to cache?

**Review by ShakedZrihen — APPROVED** [review]

File: action.yml (hunk 1 above)
Comment (MishaKav) [reply action.yml:114]: it will cache all dockers... good question. I will think about it.

Comment (MishaKav) [reply action.yml:114]: It will cache all docker-layers inside **this** workflow, so actually, since we use only `rules-engine` - it will cache only him.

**Comment by jordanjones98** [comment]
This PR seems to have broke things for people that don't allow non verified actions.

Also, why was an action used that in it's README specifically says it is not maintained?

**Pull request merged**